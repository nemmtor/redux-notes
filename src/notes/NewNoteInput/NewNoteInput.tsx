import React, { ChangeEvent, useState } from 'react';

interface IProps {
  addNote: (note: string) => void;
}

const initialNoteState = '';

export const NewNoteInput: React.FC<IProps> = ({ addNote }) => {
  const [note, setNote] = useState(initialNoteState);

  const handleNoteChange = (e: ChangeEvent<HTMLInputElement>) => {
    setNote(e.target.value);
  };

  const handleNoteSubmit = () => {
    addNote(note);
    setNote(initialNoteState);
  };

  return (
    <main data-testid="note-container">
      <input
        type="text"
        name="note"
        placeholder="Note"
        onChange={handleNoteChange}
        value={note}
      />
      <button type="button" onClick={handleNoteSubmit}>
        Add note
      </button>
    </main>
  );
};
