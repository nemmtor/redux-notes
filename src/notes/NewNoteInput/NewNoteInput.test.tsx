import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { NewNoteInput } from './NewNoteInput';

import '@testing-library/jest-dom';

describe('NewNoteInput', () => {
  it('has valid text content', () => {
    const addNote = jest.fn();
    const { getByTestId } = render(<NewNoteInput addNote={addNote} />);
    expect(getByTestId('note-container')).toHaveTextContent('Add note');
  });

  it('matches snapshot', () => {
    const addNote = jest.fn();
    const tree = renderer.create(<NewNoteInput addNote={addNote} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
