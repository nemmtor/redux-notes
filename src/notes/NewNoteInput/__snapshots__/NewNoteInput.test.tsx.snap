// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewNoteInput matches snapshot 1`] = `
<main
  data-testid="note-container"
>
  <input
    name="note"
    onChange={[Function]}
    placeholder="Note"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Add note
  </button>
</main>
`;
